pallets/nft/src/lib.rs

```diff
@@ -2,277 +2,218 @@
-			orml_nft::Pallet::<T>::burn(&sender, token)?;
-			Self::deposit_event(Event::TokenBurned(sender, token.0, token.1));
-			Ok(())
-		}
-
-		/// Removes a class from existence
-		/// Returns the bond amount
-		///
-		/// Parameters:
-		/// - `class_id`: unique identificator of a class
-		#[pallet::weight(<T as Config>::WeightInfo::destroy_class())]
-		#[transactional]
-		pub fn destroy_class(origin: OriginFor<T>, class_id: ClassIdOf<T>) -> DispatchResult {
+		pub fn burn(origin: OriginFor<T>, class_id: T::NftClassId, instance_id: T::NftInstanceId) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
-			let class_info = orml_nft::Pallet::<T>::classes(class_id).ok_or(Error::<T>::ClassNotFound)?;
-			ensure!(class_info.total_issuance == Zero::zero(), Error::<T>::NonZeroIssuance);
-			orml_nft::Pallet::<T>::destroy_class(&sender, class_id)?;
-			T::Currency::unreserve(&sender, T::ClassBondAmount::get());
-			Self::deposit_event(Event::TokenClassDestroyed(sender, class_id));
-			Ok(())
-		}
 
-		///////////////////////////////////////////////////
-		//
-		// Basilisk-specific methods for NFT handling
-		//
-		///////////////////////////////////////////////////
+			let class_type = Self::classes(class_id)
+				.map(|c| c.class_type)
+				.ok_or(Error::<T>::ClassUnknown)?;
 
-		/// Similar method to create_/destroy_class
-		/// The difference between a pool and a class in this case is that
-		/// a price has to be specified for each pool. Any NFT within this class
-		/// will have this exact constant price
-		///
-		/// Parameters:
-		/// - `metadata`: Arbitrary info/description of a pool
-		/// - `data`: Field(s) defined in the ClassData struct
-		/// - `price`: Price of each individual NFT
-		#[pallet::weight(<T as Config>::WeightInfo::create_pool())]
-		#[transactional]
-		pub fn create_pool(
-			origin: OriginFor<T>,
-			metadata: Vec<u8>,
-			data: T::ClassData,
-			price: BalanceOf<T>,
-		) -> DispatchResult {
-			let sender = ensure_signed(origin)?;
+			ensure!(T::Permissions::can_burn(&class_type), Error::<T>::NotPermitted);
 
-			T::Currency::reserve(&sender, T::ClassBondAmount::get())?;
-			let class_id = orml_nft::Pallet::<T>::create_class(&sender, metadata, data)?;
-			PoolItemPrice::<T>::insert(class_id, price);
+			Self::do_burn(sender, class_id, instance_id)?;
 
-			Self::deposit_event(Event::TokenPoolCreated(sender, class_id));
 			Ok(())
 		}
 
-		/// Removes a pool from existence
-		/// Returns the bond amount
+		/// Removes a class from existence
 		///
 		/// Parameters:
-		/// - `class_id`: unique identificator of a class
+		/// - `class_id`: The identifier of the asset class to be destroyed.
 		#[pallet::weight(<T as Config>::WeightInfo::destroy_class())]
 		#[transactional]
-		pub fn destroy_pool(origin: OriginFor<T>, class_id: ClassIdOf<T>) -> DispatchResult {
+		pub fn destroy_class(origin: OriginFor<T>, class_id: T::NftClassId) -> DispatchResultWithPostInfo {
```

Comment:
do we need `DispatchResultWithPostInfo`, why not use just a `DispatchResult`?

Reply:
For us it would be but Uniques requires it. They calculate destroy weight based on class content. Therefore we have to comply.